ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsNoJobTask.java

```diff
@@ -265,30 +351,16 @@ private int aggregateStats(ExecutorService threadPool, Hive db) {
 
       Table table = tableSpecs.tableHandle;
 
-      Collection<Partition> partitions = null;
-      if (work.getPartitions() == null || work.getPartitions().isEmpty()) {
-        if (table.isPartitioned()) {
-          partitions = tableSpecs.partitions;
-        }
-      } else {
-        partitions = work.getPartitions();
-      }
+      List<Partish> partishes = getPartishes(table);
 
-      LinkedList<Partish> partishes = Lists.newLinkedList();
-      if (partitions == null) {
-        partishes.add(Partish.buildFor(table));
+      List<StatCollector> scs;
+      if (useBasicStatsFromStorageHandler(table)) {
+        scs = partishes.stream().map(HiveStorageHandlerStatCollector::new).collect(Collectors.toList());
```

Comment:
note: I think if you use a simple loop with a conditional it will be more readable...

the current solution duplicates: `partishes.stream().map` and `.collect(Collectors.toList())`